port/main.c

```diff
@@ -132,17 +132,17 @@ static void mainLoop(void)
 	status_oid.port = port;
 	pf_oid.port = port;
 
-	if (create_dev(&route_oid, "/dev/route") < 0) {
+	if (portRegister(port, "/dev/route", &route_oid) < 0) {
```

Comment:
This change will make MMU target regress. create_dev is not working because NOMMU lacking filesystem which can handle mkdir("/dev", 0666);

We can go three ways here:
- make distinction in lwip (so use portRegister on xxx target, use create_dev on yyy target)
- handle this in libphoenix (use portRegister when mkdir failed)
- ensure that every system have some sort of mkdir capable rootfs.

@nalajcie Which way you think is best? I'm leaning to handling in libphoenix option, personally.

Reply:
I think `create_dev` is part of our standard library API so it should work on every supported platform so I would opt for always using `create_dev` and fixing the implementation in libphoenix.

To be on the same page: `mkdir` is a fallback in case `devfs` lookup fails (see https://github.com/phoenix-rtos/libphoenix/blob/6fb4fa6b3f582931ad5a336bba004cf91069e951/unistd/file.c#L219) - maybe we can provide some simple NOMMU devfs instead (e.g. without directory support?). I'm not familiar with phoenix-rtos convention - does it have to be a separate process or can it be handled differently (in-kernel? ;))?

As You suggested - the quick fix is to use `portRegister` in `create_dev` implementation - maybe we should always do it on NOMMU platforms for consistency? I don't know all of the drawbacks of this solution so I can't recommend the specific way.